project_task_stage_change_restriction/models/project_task_stage.py

```
@@ -0,0 +1,36 @@
+        "group_id",
+        string="Group Members",
+    )
+
+    def has_restrictions(self):
+        """Return True if the stage has at least one restriction."""
+        self.ensure_one()
+        return bool(
+            self.assigned_only or self.project_manager_only or self.allowed_group_ids
+        )
+
+    def _user_passes_group_rule(self, user):
```

Ruff (0.11.9)

Missing return type annotation for private function `_user_passes_group_rule`

(ANN202)